packages/infra/lib/hl7v2-stack/vpn.ts

```diff
@@ -0,0 +1,69 @@
+import * as cdk from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Construct } from "constructs";
+import { Hl7v2VpnConfig } from "../../config/hl7v2-config";
+export interface VpnStackProps extends cdk.NestedStackProps {
+  vpc: ec2.IVpc;
+  vpnConfig: Hl7v2VpnConfig;
+}
+
+export class VpnStack extends cdk.NestedStack {
+  constructor(scope: Construct, id: string, props: VpnStackProps) {
+    super(scope, id, props);
+
+    // 1. Create a Virtual Private Gateway (VGW)
+    const vpnGateway = new ec2.CfnVPNGateway(this, "VpnGateway", {
+      type: "ipsec.1",
+      amazonSideAsn: 65000, // or remove to let AWS pick a default
+    });
+
+    // Attach the VGW to your VPC
+    new ec2.CfnVPCGatewayAttachment(this, "VpcVpnAttachment", {
+      vpcId: props.vpc.vpcId,
+      vpnGatewayId: vpnGateway.ref,
+    });
+
+    // 2. Create a Customer Gateway
+    const customerGateway = new ec2.CfnCustomerGateway(this, "CustomerGateway", {
+      bgpAsn: props.vpnConfig.bgpAsn,
+      ipAddress: props.vpnConfig.customerGatewayPublicIp,
+      type: "ipsec.1",
+    });
+
+    // 3. Create the VPN Connection
+    const vpnConnection = new ec2.CfnVPNConnection(this, "VpnConnection", {
+      type: "ipsec.1",
```

Comment:
I think we'll always use the same type, so maybe just move it to a const?

Reply:
addressed